src/interaction.js

```diff
@@ -0,0 +1,43 @@
+// update todo
+
+const updateTask = (value, id) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  const newArr = todos.findIndex((elem) => elem.index === Number(id));
+  todos[newArr].desc = value;
+  localStorage.setItem('todos', JSON.stringify(todos) || []);
+};
+
+// remove item
+
+const removeTodo = (id) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  const newTodos = todos.filter((item) => item.index !== id);
+  for (let i = 0; i < newTodos.length; i += 1) {
+    newTodos[i].index = i + 1;
+  }
+  localStorage.setItem('todos', JSON.stringify(newTodos) || []);
+  window.location.reload();
+};
+
+// mark complete or uncomplete
+
+const markComplete = (id, check) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  todos.forEach((item, ind) => {
+    if (item.index === Number(id)) {
+      todos[ind].completed = check;
+    }
+  });
+  localStorage.setItem('todos', JSON.stringify(todos) || []);
+};
+// clear all completed mark
+
+const clearCompleted = () => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  const completedTodos = todos.filter((item) => item.completed !== true);
+  localStorage.setItem('todos', JSON.stringify(completedTodos) || []);
+  window.location.reload();
+};
```

Comment:
- [ ] When a user deletes all the completed tasks, the values that are remaining are supposed to be saved in the local storage **and their indexes should be updated and start from 1**. Currently, when I have three todos and I finish the first one and delete the first one, I should have an id of 1 and 2 and not 2 and 3 in my local storage. Hope this makes sense.

**This is what the project requirements say:**

![image](https://user-images.githubusercontent.com/86318284/214921435-32060110-25d7-4983-a2dd-8136fdc5d34b.png)

**This is what I mean that your id's are not updating correctly:**

[Screencast from 26-01-2023 09:39:21 ALASIRI.webm](https://user-images.githubusercontent.com/86318284/214921706-626ce20e-a10e-45a8-8da7-1124b68b8e58.webm)